test/keccak-transcript-tests.t.sol

```diff
@@ -0,0 +1,116 @@
+        uniform[30] = 0x44; uniform[31] = 0x68; uniform[32] = 0xdc; uniform[33] = 0x49; uniform[34] = 0xef; uniform[35] = 0x4f;
+        uniform[36] = 0x2b; uniform[37] = 0xc8; uniform[38] = 0x1a; uniform[39] = 0xf4; uniform[40] = 0x50; uniform[41] = 0xae;
+        uniform[42] = 0x32; uniform[43] = 0x73; uniform[44] = 0xda; uniform[45] = 0xd1; uniform[46] = 0xe9; uniform[47] = 0xe2;
+        uniform[48] = 0xce; uniform[49] = 0x0e; uniform[50] = 0x16; uniform[51] = 0xb7; uniform[52] = 0xb5; uniform[53] = 0x3f;
+        uniform[54] = 0xa2; uniform[55] = 0xd1; uniform[56] = 0x45; uniform[57] = 0x5e; uniform[58] = 0x72; uniform[59] = 0x7d;
+        uniform[60] = 0xba; uniform[61] = 0x32; uniform[62] = 0xff; uniform[63] = 0x15;
+
+
+        uint256 scalarPallas = ScalarFromUniformLib.scalarFromUniform(uniform, ScalarFromUniformLib.curvePallas());
+        uint256 expectedPallas = 0x076a01b2fad03f35458586f926ba83eeebf8c4fa3b7bcfa1573e1464e11f5358;
+        assertEq(scalarPallas, expectedPallas);
+
+        uint256 scalarVesta = ScalarFromUniformLib.scalarFromUniform(uniform, ScalarFromUniformLib.curveVesta());
+        uint256 expectedVesta = 0x0c34755d6b4566f930b8371afd2e4818ae49878a10527fd4443dbec811582d43;
+        assertEq(scalarVesta, expectedVesta);
+    }
+
+    function testKeccakTranscriptPallas() public {
+        uint8[] memory input = new uint8[](4); // b"test" in Rust
+        input[0] = 0x74;
+        input[1] = 0x65;
+        input[2] = 0x73;
+        input[3] = 0x74;
+
+        KeccakTranscriptLib.KeccakTranscript memory transcript = KeccakTranscriptLib.instantiate(input);
+
+        uint256 input1 = 123456789;
+        uint8[] memory label1 = new uint8[](2); // b"s1" in Rust
+        label1[0] = 0x73;
+        label1[1] = 0x31;
+
+        uint256 input2 = 987654321;
+        uint8[] memory label2 = new uint8[](2); // b"s2" in Rust
+        label2[0] = 0x73;
+        label2[1] = 0x32;
+
+        KeccakTranscriptLib.KeccakTranscript memory transcript1 = KeccakTranscriptLib.absorb(transcript, label1, input1);
+        KeccakTranscriptLib.KeccakTranscript memory transcript2 = KeccakTranscriptLib.absorb(transcript1, label2, input2);
+
+        uint8[] memory squeezeLabel = new uint8[](2); // b"c1" in Rust
+        squeezeLabel[0] = 0x63;
+        squeezeLabel[1] = 0x31;
+
+        ScalarFromUniformLib.Curve curve = ScalarFromUniformLib.curvePallas();
+        (, uint256 output) = KeccakTranscriptLib.squeeze(transcript2, curve, squeezeLabel);
+
+        uint256 expected = 0x3484d00943072ca9baa6dcd107e9bf09d599558b2a8b37ebd9e50e340c5d03a3;
+        assertEq(output, expected);
+    }
+
+    // Following test has been ported: https://github.com/microsoft/Nova/blob/main/src/provider/keccak.rs#L107
+    // TODO: For some reason, test vector passes if using Vesta curve parameters, but in reference implementation type of point is Pallas
```

Comment:
I think the reason the test is passing with Vesta rather than Pallas is because they're taking `G = pasta_curves::pallas::Point`, so the scalar field over which all the arithmetic is being done is the Vesta base field.